content/post/2023-08-09-data-transfer.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: post
+date: 2023-08-09
+title: Data transfer at scale(Part 2)
+author: fink
+tags: [mining]
+---
+
+In the [previous post](https://fink-broker.org/2023-01-17-data-transfer), we announced a new service of Fink to explore and transfer historical data at scale from the website [https://fink-portal.org/download](https://fink-portal.org/download). And now we'll be adding features to reduce the download time and processing time from the user's machine.
+<!---->
+
+### What's new
+
+In this version we have introduced the functionality of simultaneous downloading from multiple partitions through the implementation of multi-processing technology an approach that takes advantage of modern hardware resources to run multiple tasks in parallel. 
+By using this strategy, the service is able to simultaneously access different partitions of the data stored in the Kafka server, enabling faster and more efficient transfer. The benefits of this approach are numerous, ranging from optimizing transfer times to making more efficient use of available hardware resources.
+
+### How to use
+
+The basic command for downloading data from the Fink Apache Kafka cluster includes the following options :
+1. `topic` : This is the name of the topic where alerts are loaded.
+2. `outdir` : The directory where the alerts will be stored.
+3. `partitionby` : Specifies how the alerts will be partitioned.
+
+```bash
+fink_datatransfer \
+            -topic <topic name> \
+            -outdir <output directory> \
+            -partitionby time \
+            --verbose
+```
+In this new version, other options will need to be taken into account:
+1. `nconsumers` : Determine the number of processes (consumers) to use.\
+Each process will be assigned a partition in which it will consume all available alerts before being assigned a new, either partially processed or unprocessed partition.
+For example, with 10 partitions in our topic and 4 consumers, the first consumer will be assigned the first partition, the second consumer the second partition, and so on. Partition 5 will be assigned to the first one to finish retrieving all available alerts in its partition.
+In our examples, we used a computer with 8 cores. The program began experiencing crashes after using a number of consumers greater than 6. Therefore, it is recommended not to use a number of consumers higher than the available number of cores.
```

Comment:
Can you elaborate more what you mean by `crash` in this thread? What kind of errors this was?